### PR TITLE
feat: added support for trailing spaces in pipeline run command

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description

To start workflow, `/run pipeline` command is also supported to include a space in the command i.e. `/run pipeline `
This PR addresses [Issue-9252](https://github.ibm.com/GoldenEye/issues/issues/9252) and [Issue-9346](https://github.ibm.com/GoldenEye/issues/issues/9346)

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content
- Enhanced the workflow trigger logic to consider a trailing space in the pipeline execution command.
- Improved error logging for `Notify-Contributor-Requirements` Job.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by imprateeksh** [comment]
workflow execution happens only using the main branch's workflow file, hence the job skips if I try to provide `/run pipeline ` i.e a space at last

File: .github/workflows/common-terraform-module-ci-v2.yml [hunk 1]
```diff
@@ -59,10 +59,17 @@ jobs:
     uses: ./.github/workflows/post-usage.yml
     secrets: inherit
 
+  test_message:
```
Comment (daniel-butler-irl) [inline .github/workflows/common-terraform-module-ci-v2.yml:62]: remove this test

Comment (imprateeksh) [reply .github/workflows/common-terraform-module-ci-v2.yml:62]: Sure, thank you for pointing out. 
As part of trying to verify I might have missed this.Will remove this.

**Comment by daniel-butler-irl** [comment]
Still not working https://github.com/terraform-ibm-modules/mock-module/pull/728
https://github.com/terraform-ibm-modules/mock-module/actions/runs/9595140924/job/26459205744
image

File: .github/workflows/post-usage.yml [hunk 2]
```diff
@@ -9,23 +9,44 @@ on:
...
-          "https://api.github.com/orgs/${{github.event.organization.login}}/teams/${{inputs.required_team}}/memberships/${{github.actor}}" | jq -r '.state')
+
+          API_URL="https://api.github.com/orgs/${{github.event.organization.login}}/teams/${{inputs.required_team}}/memberships/${{github.actor}}"
+          echo "Fetching the response for $API_URL"
+
+          api_response=$(curl -s -o response_body.json -w "%{http_code}" -X GET -H "Authorization: Bearer ${{secrets.ORG_READER_GH_TOKEN}}" \
+          -H "Accept: application/vnd.github.v3+json" "$API_URL")
+
+          # Log messages if API failure is observed
+          if [[ "$api_response" -ne 200 && "$api_response" -ne 201 ]]; then
```
Comment (daniel-butler-irl) [inline .github/workflows/post-usage.yml:29]: If as user is not found there is a 404 response. We want the message to be printed if the user is not found. I think it should print the message on any failure.

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]

**Comment by imprateeksh** [comment]
> Still not working [terraform-ibm-modules/mock-module#728](https://github.com/terraform-ibm-modules/mock-module/pull/728) https://github.com/terraform-ibm-modules/mock-module/actions/runs/9595140924/job/26459205744 image
Thanks for verifying Daniel. I am making a fix to show the error message but I think the `/run pipeline ` (with a space at last) is working because the workflow progressed and entered the if block.

**Comment by imprateeksh** [comment]
@daniel-butler-irl  I have made the changes, explicitly checking for 404 and other errors. Requesting your review.  I tried running the same in your PR - https://github.com/terraform-ibm-modules/mock-module/actions/runs/9608908483

**Comment by daniel-butler-irl** [comment]
You still have not tested when a user is not in the required team. It does not post the message as it did before even when using the correct syntax without the space.

**Comment by imprateeksh** [comment]
@daniel-butler-irl - I have tested the changes [here](https://github.com/terraform-ibm-modules/mock-module/pull/727#issuecomment-2186595344).
Had to remove the exit statements in order to proceed to the next step.

**(a) With a user who is not a member:** Workflow run [here](https://github.com/terraform-ibm-modules/mock-module/actions/runs/9646321827/job/26602475681)
image

image

**(b) With a member user:** Workflow run [here](https://github.com/terraform-ibm-modules/mock-module/actions/runs/9646458679/job/26602915623)
image

File: .github/workflows/post-usage.yml (hunk 2 above)
Comment (imprateeksh) [reply .github/workflows/post-usage.yml]: Thanks for your input. I have made the changes to print error messages.

**Review by daniel-butler-irl — APPROVED** [review]

**Comment by daniel-butler-irl** [comment]
@imprateeksh As soon as the new release is generated please open a PR against the mock module to use the new version, so we don't forget about it. Thanks

**Comment by imprateeksh** [comment]
> @imprateeksh As soon as the new release is generated please open a PR against the mock module to use the new version, so we don't forget about it. Thanks

[PR](https://github.com/terraform-ibm-modules/mock-module/pull/732) opened in mock-module.

**Pull request merged**